File: board_dsl/src/main/java/com/dooho/board/entity/UserEntity.java
package com.dooho.board.entity;

import com.dooho.board.dto.auth.SignUpDto;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name="User")
@Table(name="User")
//Entity(name="") -> 해당 클래스를 Entity 클래스로 사용하겠다.
//Table(name="") db에 있는 해당 테이블과 클래스를 매핑시킨다.
public class UserEntity {
    @Id
    private String userEmail;
    private String userPassword;
    private String userNickname;
    private String userPhoneNumber;
    private String userAddress;
    private String userProfile;

    public UserEntity(SignUpDto dto){
        this.userEmail = dto.getUserEmail();
        this.userPassword = dto.getUserPassword();
        this.userNickname = dto.getUserNickname();
        this.userPhoneNumber = dto.getUserPhoneNumber();
        this.userAddress = dto.getUserAddress();
        this.userProfile = dto.getUserProfile();
    }

}

File: board_spring/src/main/java/com/dooho/board/entity/UserEntity.kt
package com.dooho.board.entity

import com.dooho.board.dto.auth.SignUpDto
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "User")
data class UserEntity(
    @Id
    var userEmail: String?,
    var userPassword: String?,
    var userNickname: String?,
    var userPhoneNumber: String?,
    var userAddress: String?,
    var userProfile: String?
) {
    constructor(dto: SignUpDto?) : this(
        dto?.userEmail,
        dto?.userPassword,
        dto?.userNickname,
        dto?.userPhoneNumber,
        dto?.userAddress,
        dto?.userProfile
    )

    constructor() : this(null, null, null, null, null, null)


    // equals, hashCode, toString 메서드는 자동으로 생성됨
}